contracts/secret-storage/SecretStore.sol

```diff
@@ -365,7 +365,9 @@ contract SecretStore is
         uint256[] memory stakeAmounts = TEE_MANAGER.getTeeNodesStake(_enclaveAddresses);
         uint256 len = _enclaveAddresses.length;
         for (uint256 index = 0; index < len; index++) {
-            _insert_unchecked(_env, _enclaveAddresses[index], uint64(stakeAmounts[index] / STAKE_ADJUSTMENT_FACTOR));
+            address enclaveAddress = _enclaveAddresses[index];
+            if (secretStores[enclaveAddress].storageOccupied < secretStores[enclaveAddress].storageCapacity)
```

Comment:
make sure whoever is calling this method, doesn't put this check of storageOccupied, otherwise it would be redundant.